be/src/cache/datacache.cpp

```diff
@@ -173,9 +189,20 @@ StatusOr<CacheOptions> DataCache::_init_cache_options() {
             // we do not automatically rename it when the source and destination directories are on different disks.
             // In this case, users should manually remount the directories and restart them.
             std::string datacache_path = root_path.path + "/datacache";
-            std::string starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
 #ifdef USE_STAROS
             if (config::datacache_unified_instance_enable) {
+                std::string starlet_cache_path;
+                if (idx < corresponding_starlet_dirs.size()) {
+                    if (!corresponding_starlet_dirs[idx].empty()) {
+                        starlet_cache_path = corresponding_starlet_dirs[idx] + "/star_cache";
+                    } else {
+                        // empty means we can't find an old starlet cache dir for this root_path, use the default one
+                        starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
+                    }
+                    idx++;
+                } else {
+                    starlet_cache_path = root_path.path + "/starlet_cache/star_cache";
+                }
```

Comment:
It seems assign the same concatenated string twice is redundant and error-prone. You can directly use `root_path.path + "/starlet_cache/star_cache"` to initialize the default value, and then adjust it to `orresponding_starlet_dirs[idx] + "/star_cache"`only when necessary.